src/conn/conn_prefetch.c

```diff
@@ -106,6 +106,15 @@ __wt_prefetch_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
         locked = false;
 
         WT_WITH_DHANDLE(session, pe->dhandle, ret = __wt_prefetch_page_in(session, pe));
+
+        /*
+         * It's a weird case, but if verify is utilizing prefetch and encounters a corrupted block,
+         * stop using prefetch. Some of the guarantees about ref and page freeing are ignored in
+         * that case, which can invalidate entries on the prefetch queue.
+         */
+        if (!F_ISSET(S2C(session), WT_CONN_DATA_CORRUPTION))
```

Comment:
Could we use the `conn` variable instead of doing `S2C(session)`?

Reply:
That's true!  [37ce7ae](https://github.com/wiredtiger/wiredtiger/pull/9929/commits/37ce7ae8b0971641f1f4ce915487f850f1d90143)